tests/unittests/wrappers/test_transformations.py

```diff
@@ -0,0 +1,234 @@
+
+    def test_no_base_metric(self) -> None:
+        """Tests that TypeError is raised when no wrapped_metric is passed."""
+        with pytest.raises(TypeError, match=r"Expected wrapped metric to be an instance of .*"):
+            MetricInputTransformer([])
+
+
+class TestLambdaInputTransformer:
+    """Test suite for LambdaInputTransformer."""
+
+    _test_signature: ClassVar = (
+        "cls",
+        "transform_pred",
+        "transform_target",
+        "preds",
+        "preds_transformed",
+        "targets",
+        "targets_transformed",
+    )
+    _test_cases: ClassVar = [
+        # No change to input data (identity transform)
+        (
+            BinaryAccuracy,
+            None,
+            None,
+            [0.9, 0.8, 0.7, 0.6, 0.5, 0.6, 0.7, 0.8, 0.5, 0.4],
+            [0.9, 0.8, 0.7, 0.6, 0.5, 0.6, 0.7, 0.8, 0.5, 0.4],
+            [1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0],
+        ),
+        # Change to pred data (invert transform)
+        (
+            BinaryAccuracy,
+            lambda pred: 1 - pred,
+            None,
+            [0.9, 0.8, 0.7, 0.6, 0.5, 0.6, 0.7, 0.8, 0.5, 0.4],
+            [0.1, 0.2, 0.3, 0.4, 0.5, 0.4, 0.3, 0.2, 0.5, 0.6],
+            [1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0],
+        ),
+        # Change to target data (invert transform)
+        (
+            BinaryAccuracy,
+            None,
+            lambda target: 1 - target,
+            [0.9, 0.8, 0.7, 0.6, 0.5, 0.6, 0.7, 0.8, 0.5, 0.4],
+            [0.9, 0.8, 0.7, 0.6, 0.5, 0.6, 0.7, 0.8, 0.5, 0.4],
+            [1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0],
+            [0.0, 1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 1.0, 1.0, 1.0],
+        ),
+        # Change to all input data (invert transform)
+        (
+            BinaryAccuracy,
+            lambda pred: 1 - pred,
+            lambda target: 1 - target,
+            [0.9, 0.8, 0.7, 0.6, 0.5, 0.6, 0.7, 0.8, 0.5, 0.4],
+            [0.1, 0.2, 0.3, 0.4, 0.5, 0.4, 0.3, 0.2, 0.5, 0.6],
+            [1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0],
+            [0.0, 1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 1.0, 1.0, 1.0],
+        ),
+    ]
+
+    @pytest.mark.parametrize(_test_signature, _test_cases)
+    def test_forward(
+        self, cls, transform_pred, transform_target, preds, preds_transformed, targets, targets_transformed
+    ) -> None:
+        """Tests if the binarized forward matches the output of the metric on manually binarized targets."""
+        metric = cls()
+        wrapped_metric = LambdaInputTransformer(
+            metric, transform_pred=transform_pred, transform_target=transform_target
+        )
+        preds = Tensor(preds).float()
+        targets = Tensor(targets).float()
+        preds_transformed = Tensor(preds_transformed).float() if preds_transformed is not None else preds
```

Comment:
`preds_transformed` and `targets_transformed` are never None, so these statements can be removed:
```suggestion
        preds_transformed = Tensor(preds_transformed).float()
```